File: patterns-java/src/test/java/me/freedom4live/patterns/java/bridge/CarTest.java
package me.freedom4live.patterns.java.bridge;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

class CarTest {

    private Engine dieselEngine;
    private Engine reactiveEngine;


    @BeforeEach
    void setUp() {
        dieselEngine = new DieselEngine();
        reactiveEngine = new ReactiveEngine();
    }

    @Test
    void drive_dieselEngine_500power() {
        //arrange
        Car dieselCar = new Car(dieselEngine);

        //act
        int resultPower = dieselCar.drive();

        //assert
        assertEquals(500, resultPower);
    }

    @Test
    void drive_reactiveEngine_1000Power() {
        //arrange
        Car reactiveCar = new Car(reactiveEngine);

        //act
        int resultPower = reactiveCar.drive();

        //assert
        assertEquals(1000, resultPower);
    }

}

File: patterns-kotlin/src/test/kotlin/me/freedom4live/patterns/kotlin/bridge/CarTest.kt
package me.freedom4live.patterns.kotlin.bridge

import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

internal class CarTest {

    private lateinit var dieselEngine: Engine
    private lateinit var reactiveEngine: Engine


    @BeforeEach
    fun setUp() {
        dieselEngine = DieselEngine()
        reactiveEngine = ReactiveEngine()
    }

    @Test
    fun drive_dieselCar_500power() {
        //assert
        val dieselCar = Car(dieselEngine)

        //act
        val resultPower = dieselCar.drive()

        //assert
        Assertions.assertEquals(500, resultPower)
    }

    @Test
    fun drive_reactiveCar_1000power() {
        //assert
        val reactiveCar = Car(reactiveEngine)

        //act
        val resultPower = reactiveCar.drive()

        //assert
        Assertions.assertEquals(1000, resultPower)
    }

}
